test/e2e/e2e_test.go

```diff
@@ -955,6 +955,248 @@ var _ = Describe("e2e multiple external gateway update validation", func() {
+		if haMode {
+			// ha ci mode runs a named set of nodes with a prefix of ovn-control-plane
+			ciWorkerNodeSrc = ovnHaWorkerNode
+		}
+		nodeIP, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("failed to get the node ip address from node %s %v", ciWorkerNodeSrc, err)
+		}
+		nodeIP = strings.TrimSuffix(nodeIP, "\n")
+		if ip := net.ParseIP(nodeIP); ip == nil {
+			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", ciWorkerNodeSrc, nodeIP)
+		}
+		framework.Logf("the pod side node is %s and the source node ip is %s", ciWorkerNodeSrc, nodeIP)
+		podCIDR, err := getNodePodCIDR(ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("Error retrieving the pod cidr from %s %v", ciWorkerNodeSrc, err)
+		}
+		framework.Logf("the pod cidr for node %s is %s", ciWorkerNodeSrc, podCIDR)
+		// add loopback interface used to validate all traffic is getting drained through the gateway
+		_, err = runCommand("docker", "exec", gwContainerNameAlt1, "ip", "address", "add", exGWRemoteCidrAlt1, "dev", "lo")
+		if err != nil {
+			framework.Failf("failed to add the loopback ip to dev lo on the test container: %v", err)
+		}
+		// Create the pod that will be used as the source for the connectivity test
+		createGenericPod(f, srcPingPodName, ciWorkerNodeSrc, command)
+
+		// Wait for pod exgw setup to be almost ready
+		wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+			jsonFlag := "jsonpath='{.metadata.annotations.k8s\\.ovn\\.org/routing-external-gws}'"
+			kubectlOut, err := framework.RunKubectl("get", "pod", ovnNsFlag, srcPingPodName, jsonFlag)
+			if err != nil {
+				return false, nil
+			}
+			if !strings.Contains(kubectlOut, exGWRemoteIpAlt1) {
+				return false, nil
+			}
+			return true, nil
+		})
+
+		// There is a condition with e2e WaitForPodNotPending that returns ready
+		// before calling for the IP address will succeed. This simply adds some retries.
+		for i := 1; i < getPodIPRetry; i++ {
+			pingSrc, err = getPodAddress(srcPingPodName, f.Namespace.Name)
+			if err != nil {
+				framework.Logf("Warning unable to query the test pod on node %s %v", ciWorkerNodeSrc, err)
+			}
+			validIP = net.ParseIP(pingSrc)
+			if validIP != nil {
+				framework.Logf("Source pod is %s is %s", srcPingPodName, pingSrc)
+				break
+			}
+			time.Sleep(time.Second * 4)
+			framework.Logf("Retry attempt %d to get pod IP from initializing pod %s", i, srcPingPodName)
+		}
+		// Fail the test if no address is ever retrieved
+		if validIP == nil {
+			framework.Failf("Warning: Failed to get an IP for the source pod %s, test will fail", srcPingPodName)
+		}
+		time.Sleep(time.Second * 15)
+		// Verify the gateway and remote address is reachable from the initial pod
+		By(fmt.Sprintf("Verifying connectivity without vxlan to the updated annotation and initial external gateway %s and remote address %s", exGWRemoteIpAlt1, exGWIpAlt1))
```

Comment:
should be "initial external gateway %s and remote address %s", exGWIpAlt1, exGWRemoteIpAlt1" ?